src/finn/custom_op/fpgadataflow/matrixvectoractivation.py

```diff
@@ -728,6 +751,43 @@ def get_hls_compatible_threshold_tensor(self, orig_thres_matrix):
         rows between PEs is not as expected (n_thres_steps)"""
         return ret.reshape(1, pe, tmem, n_thres_steps)
 
+    def get_hls_compatible_weight_tensor(self, orig_weight_matrix):
```

Comment:
I see that you first deleted and then added this function again. For me this looks like an HLS-specific function and thus should be moved to `matrixvectoractivation_hls.py`, was there a reason to bring it back into the hw abstraction layer?

Reply:
I think the name of the function is misleading. It suggests it's exclusively something that belongs to the HLS custom-op, but it's actually used in the `make_weight_file` method. That one is used both for the HLS and RTL custom-op and hence moved to the HW abstraction layer.
A more appropriate name would be: `get_hw_compatible_weight_tensor`. If you agree, I'll go on and rename that now.